File: atcoder.jp/contests/abs/tasks/abc085_c/main.py
def input_int_list() -> "list[int]":
    return list(map(int, input().split()))


def main() -> None:
    N, Y = input_int_list()

    ans = next(
        (
            (i, j, N - i - j)
            for i in range(N + 1)
            for j in range(N - i + 1)
            if 10000 * i + 5000 * j + 1000 * (N - i - j) == Y
        ),
        (-1, -1, -1),
    )

    print(*ans)


main()

File: atcoder.jp/contests/abs/tasks/abc085_c/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const range = (a: number, b?: number) =>
  b ? [...Array(b - a)].map((_, i) => a + i) : [...Array(a)].map((_, i) => i);

// input functions -------------------------------------------------------------
const inputNumberArray = () => input().split(" ").map(Number);
// input functions -------------------------------------------------------------

const main = () => {
  const [N, Y] = inputNumberArray();

  const solve = (): [number, number, number] => {
    for (const i of range(N + 1))
      for (const j of range(N - i + 1)) {
        const k = N - i - j;
        if (10000 * i + 5000 * j + 1000 * k === Y) return [i, j, k];
      }
    return [-1, -1, -1];
  };

  const ans = solve();
  print(...ans);
};

main();
